VikingQuiz.Api/Controllers/QuestionsController.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using VikingQuiz.Api.Models;
+using VikingQuiz.Api.Repositories;
+using VikingQuiz.Api.ViewModels;
+using VikingQuiz.Api.Mappers;
+using VikingQuiz.Api.Utilities;
+
+
+// For more information on enabling MVC for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace VikingQuiz.Api.Controllers
+{
+    [Route("api/[controller]")]
+    public class QuestionsController : Controller
+    {
+        private readonly QuestionRepository questionRepo;
+        private readonly AnswerRepository answerRepo;
+        private IEntityMapper<Question, QuestionViewModel> entityToVmMapper;
+        private IEntityMapper<QuestionViewModel, Question> vmToEntityMapper;
+        private IEntityMapper<Answer, AnswerViewModel> answerToVmMapper;
+        private IEntityMapper<AnswerViewModel, Answer> vmToAnswerMapper;
+
+        public QuestionsController(QuestionRepository questionRepo, AnswerRepository answerRepo, IEntityMapper<Question, QuestionViewModel> entityToVmMapper, IEntityMapper<QuestionViewModel, Question> vmToEntityMapper, IEntityMapper<Answer, AnswerViewModel> answerToVmMapper, IEntityMapper<AnswerViewModel, Answer> vmToAnswerMapper)
+        {
+            this.questionRepo = questionRepo;
+            this.answerRepo = answerRepo;
+            this.vmToEntityMapper = vmToEntityMapper;
+            this.entityToVmMapper = entityToVmMapper;
+            this.vmToAnswerMapper = vmToAnswerMapper;
+            this.answerToVmMapper = answerToVmMapper;
+        }
+
+        [HttpGet]
+        public IActionResult GetAll([FromQuery(Name = "quiz")] int quizId)
+        {
+            var questions = questionRepo.GetAllByQuizId(quizId);
+            var result = questions
+                        .Select(question => {
```

Comment:
This select does not look very nice. Please avoid using return statements inside LINQ.